ext/sqlite3/sqlite3.c

```diff
@@ -1948,6 +1949,7 @@ PHP_METHOD(SQLite3Result, fetchArray)
 			}
 
 			array_init(return_value);
+			zend_hash_real_init_packed(Z_ARRVAL_P(return_value));
```

Comment:
This should be removed, the loop below can also add associative keys in some cases

Reply:
ah right ...missed it